### DISCUSSION TITLE
Quick note on 1.8.7-p374 and Tk on OS X

### DISCUSSION THREAD

**Discussion by lloeki** [postmodern/ruby-install#494] [Show and tell]
Installs just fine on Yosemite with:

```
brew install gcc
ruby-install ruby 1.8.7-p374 -- --without-tk CC=gcc-4.9

# optionally, don't forget rubygems
cd ~/src
curl -O http://production.cf.rubygems.org/rubygems/rubygems-2.4.2.tgz
tar xvfz rubygems-2.4.2.tgz
cd rubygems-2.4.2
ruby setup.rb
```

Otherwise it complains about Tk headers (IIRC, it was the same in Mavericks). Alternative is to install XQuartz or something, but hey, this one is quicker.

I don't know if it's worthwhile to add a note about it in the README since it's totally unsupported upstream, but sadly some people still have a few projects incompatible with 1.9+.

**Comment by postmodern** [comment]
Before ruby-install, we maintained some [manual install instructions](https://github.com/postmodern/chruby/wiki/Ruby). Maybe we should maintain a Wiki of installation issues and Caveats?

**Comment by lloeki** [comment]
Good idea, as I ended up creating an issue mostly to document and share this.

`ruby-build` has a [troubleshooting section](https://github.com/sstephenson/ruby-build/wiki#troubleshooting) but it's a bit of a hodgepodge in the middle of a huge page, and the subsection titles are a bit of a mess. Should this become a thing, I suggest creating a wiki page dedicated to special case installation instructions and workarounds.

**Comment by aeberlin** [comment]
:+1:
